cwl/src/main/scala/cwl/WorkflowStep.scala

```diff
@@ -108,13 +108,8 @@ case class WorkflowStep(
        */
       def foldStepInput(currentFold: Checked[WorkflowStepInputFold], workflowStepInput: WorkflowStepInput): Checked[WorkflowStepInputFold] = currentFold flatMap {
         fold =>
-          // The source from which we expect to satisfy this input (output from other step or workflow input)
-          // TODO: this can be None, a single source, or multiple sources. Currently assuming it's a single one
-          val inputSource: String = workflowStepInput.source.flatMap(_.select[String]).get
-
-          // Name of the step input
-
-          val accumulatedNodes = fold.generatedNodes ++ knownNodes
+          // The sources from which we expect to satisfy this input (output from other step or workflow input)
+          val inputSources: List[String] = workflowStepInput.source.toList.flatMap(_.fold(WorkflowStepInputSourceToStrings))
```

Comment:
Can we add a comment to say that we're about to go through the list and for each source make sure we either have or generate the nodes they're referencing ?

Reply:
The comment might be better somewhere else but it'll help (at least me) next time I look at this to understand what we're doing with all those sources.